cake_fuzzer.py

```diff
@@ -280,27 +287,28 @@ async def start_others() -> None:
             only_paths_with_prefix=settings.only_paths_with_prefix,
             exclude_pattern=settings.exclude_paths,
         )
-        print(f"generated {len(paths)} paths")
+        total_paths = sum(len(paths[php_file]) for php_file in paths)
+        print(f"discovered {len(paths)} files to scan with total of {total_paths} paths")
 
         app_info = AppInfo(settings.webroot_dir)
         log_paths = await app_info.log_paths
         cake_path = await app_info.cakephp_path
 
-        # paths = ["Cerebrates/index"]
-
         for definition in defs:
-            attacks = [
-                AttackScenario(
-                    cake_path=cake_path,
-                    webroot_file=str(settings.index_php),
-                    strategy_name=definition.strategy_name,
-                    payload=payload,
-                    path=path,
-                    total_iterations=32,
-                    payload_guid_phrase=settings.payload_guid_phrase,
-                )
-                for payload, path in itertools.product(definition.scenarios, paths)
-            ]
+            attacks = []
+            for php_file in paths:
+                attacks += [
+                    AttackScenario(
+                        cake_path=cake_path,
+                        webroot_file=str(php_file),
+                        strategy_name=definition.strategy_name,
+                        payload=payload,
+                        path=path,
+                        total_iterations=32,
+                        payload_guid_phrase=settings.payload_guid_phrase,
+                    )
+                    for payload, path in itertools.product(definition.scenarios, paths[php_file])
+                ]
```

Comment:
You can just extend the `product`. Just make sure the formatting is ok, I just edited in here so it might be off.
```suggestion
            attacks = [
                AttackScenario(
                        cake_path=cake_path,
                        webroot_file=str(php_file),
                        strategy_name=definition.strategy_name,
                        payload=payload,
                        path=path,
                        total_iterations=32,
                        payload_guid_phrase=settings.payload_guid_phrase,
                    )
                    for payload, path, php_file in itertools.product(definition.scenarios, paths[php_file], paths)
            ]
```

Reply:
But here, product doesn't know what is the `php_file` var. I don't know how to iterate over paths and then inside paths with one `product` call.

Reply:
Makes sense. It's ok either way 🚀  :)